src/porepy/numerics/ad/operators.py

```diff
@@ -649,11 +650,48 @@ def _identify_subtree_discretizations(self, discr: list) -> list:
 
     ### Operator parsing ---------------------------------------------------------------
 
+    def value(
+        self, system_manager: pp.ad.EquationSystem, state: Optional[np.ndarray] = None
+    ) -> numbers.Real | np.ndarray | sps.spmatrix:
+        return self._evaluate(system_manager, state=state, evaluate_jacobian=False)
+
+    def value_and_jacobian(
+        self, system_manager: pp.ad.EquationSystem, state: Optional[np.ndarray] = None
+    ) -> AdArray:
+        ad = self._evaluate(system_manager, state=state, evaluate_jacobian=True)
+
+        # Casting the result to AdArray or raising an error.
+        if isinstance(ad, numbers.Real):
+            # AdArray requires 1D numpy array as value, not a scalar.
+            ad = np.array(ad).reshape(1)
+
+        if isinstance(ad, np.ndarray) and len(ad.shape) == 1:
+            return AdArray(ad, sps.csr_matrix((ad.shape[0], system_manager.num_dofs())))
+        elif isinstance(ad, (sps.spmatrix, np.ndarray)):
+            # this case coverse both, dense and sparse matrices returned from
+            # discretizations f.e.
+            raise NotImplementedError(
+                f"The Jacobian of {type(ad)} is not implemented because it is "
+                "multidimensional"
+            )
+        else:
+            return ad
+
     def evaluate(
         self,
         system_manager: pp.ad.EquationSystem,
         state: Optional[np.ndarray] = None,
-    ):  # TODO ensure the operator always returns an AD array
+    ):
```

Comment:
@IvarStefansson @OmarDuran: Do you agree with raising an error here? The alternative is to add a deprecation warning.